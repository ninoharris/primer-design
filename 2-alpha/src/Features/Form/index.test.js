import React from 'react'
import { shallow } from 'enzyme'

import { Form } from './index'

test('Form renders correctly with no inputs when no props are given', () => {
  const wrapper = shallow(<Form />)
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('input').length).toBe(4)
})

describe('Form correctly manages input displays and actions', () => {
  let FV, FG, RV, RG // random
  let updateInput, attemptCompletion
  beforeEach(() => {
    FV = 'ATGC'
    FG = 'AAAAAA'
    RV = 'TTTTTT'
    RG = 'GCTAGCTAG' // random

    updateInput = jest.fn()
    attemptCompletion = jest.fn()
  })

  test('Form displays correct inputs', () => {
    const wrapper = shallow(<Form FV={FV} FG={FG} RV={RV} RG={RG} updateInput={updateInput} attemptCompletion={attemptCompletion} />)
    expect(wrapper.find('input[name="FV"]').props().value).toBe(FV)
    expect(wrapper.find('input[name="FG"]').props().value).toBe(FG)
    expect(wrapper.find('input[name="RV"]').props().value).toBe(RV)
    expect(wrapper.find('input[name="RG"]').props().value).toBe(RG)
  })
  
  test('Form causes updateInput to be called upon changing an input', () => {
    const wrapper = shallow(<Form FV={FV} FG={FG} RV={RV} RG={RG} updateInput={updateInput} attemptCompletion={attemptCompletion} />)
    expect(updateInput.mock.calls.length).toBe(0)
    wrapper.find('input[name="FV"]').simulate('change', {
      target: {
        name: 'FV',
        value: 'HELLO WORLD'
      }
    })
    expect(updateInput.mock.calls.length).toBe(1)
    expect(updateInput.mock.calls[0][0]).toBe('FV')
    expect(updateInput.mock.calls[0][1]).toBe('HELLO WORLD')
  })

})