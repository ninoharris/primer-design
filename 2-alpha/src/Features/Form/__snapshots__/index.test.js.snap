// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly with no inputs when no props are given 1`] = `
<form
  autoComplete="prevent-autoComplete"
  className="form-group primer-form"
  onSubmit={[Function]}
>
  <div
    className="text-center"
  >
    <strong>
      Forward Primer
    </strong>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <div
        className="input-group-text"
      >
        5'
      </div>
    </div>
    <input
      autoComplete="prevent-autoComplete"
      className="form-control FV "
      name="FV"
      onChange={[Function]}
      type="text"
    />
    <input
      autoComplete="prevent-autoComplete"
      className="form-control FG"
      name="FG"
      onChange={[Function]}
      type="text"
    />
    <div
      className="input-group-append"
    >
      <div
        className="input-group-text"
      >
        3'
      </div>
    </div>
  </div>
  <Connect(PrimerPreviewSmall)
    strand="forward"
  />
  <div
    className="text-center"
  >
    <strong>
      Reverse Primer
    </strong>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <div
        className="input-group-text"
      >
        5'
      </div>
    </div>
    <input
      autoComplete="prevent-autoComplete"
      className="form-control RV"
      name="RV"
      onChange={[Function]}
      type="text"
    />
    <input
      autoComplete="prevent-autoComplete"
      className="form-control RG"
      name="RG"
      onChange={[Function]}
      type="text"
    />
    <div
      className="input-group-append"
    >
      <div
        className="input-group-text"
      >
        3'
      </div>
    </div>
  </div>
  <Connect(PrimerPreviewSmall)
    strand="reverse"
  />
  <button
    className="btn btn-primary mr-3"
    type="submit"
  >
    Submit
  </button>
  <button
    className="btn btn-info btn-small"
    type="button"
  >
    Preview reverse primer
  </button>
</form>
`;
